cirq/ops/raw_types.py

```diff
@@ -153,6 +153,48 @@ def _json_dict_(self):
         return protocols.obj_to_dict_helper(self, ['qubit', 'dimension'])
 
 
+class Tag:
+    """A wrapper around string that can be user to differentiate gates.
+
+    This can be instantiated using a string value or sub-classed to
+    differentiate or tag specific instances of gates.
+
+    Tags can then be accessed using a Gate's tag() function to allow for
+    special processing of certain instances of gates (for instance, to
+    prevent optimization or decomposing special gates).
+
+    Example:
+        >>> gate1 = cirq.XPowGate(exponent=1.0)
+        >>> gate1.add_tag(cirq.Tag('do_not_touch'))
+        >>> gate1.tags()
+        [Tag('do_not_touch')]
+        >>> class NoTouchy(cirq.Tag):
+        ...     def __repr__(self):
+        ...         return 'NoTouchy()'
+        >>> gate2 = cirq.XPowGate(exponent=1.0).add_tag(NoTouchy())
+        >>> gate2.tags()
+        [NoTouchy()]
+
+    """
+
+    def __init__(self, value: Optional[str] = None):
+        self.value = value
+
+    def __eq__(self, other):
+        return (isinstance(other, self.__class__) and
+                isinstance(self, other.__class__) and self.value == other.value)
+
+    def __str__(self):
+        return f'Tag(\'{self.value}\')'
```

Comment:
It's generally more pythonic to use `"` for the outer quotes so you don't have to escape the `'`s

Reply:
Done.